megamek/src/megamek/client/ui/swing/MegaMekGUI.java

```diff
@@ -559,68 +533,48 @@ public String getDescription() {
             }
         });
         int returnVal = fc.showOpenDialog(frame);
-        if ((returnVal != JFileChooser.APPROVE_OPTION)
-                || (fc.getSelectedFile() == null)) {
+        if ((returnVal != JFileChooser.APPROVE_OPTION) || (fc.getSelectedFile() == null)) {
             // I want a file, y'know!
             return;
         }
         HostDialog hd = new HostDialog(frame);
         hd.setVisible(true);
-        if ((hd.playerName == null) || (hd.serverPass == null) || (hd.port == 0)) {
-            return;
-        }
 
-        // Players should have to enter a non-blank, non-whitespace name.
-        boolean foundValid = false;
-        char[] nameChars = hd.playerName.toCharArray();
-        for (int loop = 0; !foundValid && (loop < nameChars.length); loop++) {
-            if (!Character.isWhitespace(nameChars[loop])) {
-                foundValid = true;
-            }
-        }
-        if (!foundValid) {
-            JOptionPane
-                    .showMessageDialog(
-                            frame,
-                            Messages.getString("MegaMek.PlayerNameAlert1.message"), Messages.getString("MegaMek.PlayerNameAlert1.title"), JOptionPane.ERROR_MESSAGE); //$NON-NLS-1$ //$NON-NLS-2$
+        if (!hd.dataValidation("MegaMek.PlayerNameAlert1.title", "MegaMek.PlayerNameAlert.message")) {
             return;
         }
 
         // kick off a RNG check
         d6();
         // start server
         try {
-            server = new Server(hd.serverPass, hd.port, hd.register, hd.register ? hd.metaserver : "");
+            server = new Server(hd.getServerPass(), hd.getPort(), hd.isRegister(), hd.isRegister() ? hd.getMetaserver() : "");
         } catch (IOException ex) {
-            System.err.println("could not create server socket on port " + hd.port);
-            String error = "Error: could not start server at localhost" + ":" + hd.port + " (" +
-                    ex.getMessage() + ").";
-            JOptionPane
-                    .showMessageDialog(
-                            frame,
-                            error,
-                            Messages.getString("MegaMek.HostGameAlert.title"), JOptionPane.ERROR_MESSAGE); //$NON-NLS-1$
+            MegaMek.getLogger().error(this, "could not create server socket on port " + hd.getPort());
+            JOptionPane.showMessageDialog(frame,
+                    "Error: could not start server at localhost" + ":" + hd.getPort() + " (" + ex.getMessage() + ").",
+                    Messages.getString("MegaMek.HostGameAlert.title"), JOptionPane.ERROR_MESSAGE);
```

Comment:
Is it intentional not to log the ex.getMessage()? Otherwise, the String error of line 596 could be re-used in 553 and 555

Reply:
Ended up finding a bunch of localization issues in this, so I've fixed that now.